plugins/org.obeonetwork.m2doc.sirius/src/org/obeonetwork/m2doc/sirius/providers/AbstractSiriusDiagramImagesProvider.java

```diff
@@ -124,9 +126,21 @@ protected Diagram getGmfDiagram(DSemanticDiagram semanticDiagram) {
             if (dRepresentation instanceof DSemanticDiagram) {
                 DSemanticDiagram dsd = (DSemanticDiagram) dRepresentation;
                 String filePath = getDiagramImageFilename(dsd, rootPath);
-                File file = new File(filePath);
-                file.getParentFile().mkdirs();
-                final IPath path = new Path(filePath);
+                final IPath path = ResourcesPlugin.getWorkspace().getRoot().getFile(new Path(filePath)).getLocation();

```

Comment:
This is not correct because we add dependencies to Eclipse plugins. And I think we want to be able to launch the generation in a standalone mod at some point.

Moreover, the computed file path just above is suppose to be the absolute location were to put the image. It is using the rootPath for that. The root path is computed in org.obeonetwork.m2doc.ui.command.GenerateHandler.generate(Generation, Map<String, Object>, Shell). And it is supposed to be an absolute path or at least this is currently a requirement of this class to be absolute.

So if the file path is not correct, it means the root path is not correct. And the problem should be treated were this root path is computed either by fixing a bug not giving the absolute path or if it is not meant to be an absolute then give it the absolute in the map.